src/PolyType.Roslyn/Model/AssociatedTypeModel.cs

```diff
@@ -8,4 +8,33 @@ namespace PolyType.Roslyn;
 /// <param name="AssociatedType">The symbol for the related type.</param>
 /// <param name="AssociatingAssembly">The assembly that declared the association.</param>
 /// <param name="Location">The location where the association is declared.</param>
-public record struct AssociatedTypeModel(INamedTypeSymbol AssociatedType, IAssemblySymbol AssociatingAssembly, Location? Location);
+/// <param name="Requirements">The associated type's requirements.</param>
+public record struct AssociatedTypeModel(INamedTypeSymbol AssociatedType, IAssemblySymbol AssociatingAssembly, Location? Location, TypeShapeRequirements Requirements);
+
+/// <summary>
+/// Describes the requirements for preparing an associated type.
+/// </summary>
+/// <devremarks>
+/// Keep this in sync with the TypeShapeRequirements enum defined in the PolyType assembly.
+/// </devremarks>
+[Flags]
+public enum TypeShapeRequirements
+{
+    /// <summary>No shape is required.</summary>
+    None = 0x0,
+
+    /// <summary>
+    /// A constructor, its parameters and their types should be included in the shape, if one is declared.
+    /// </summary>
+    Constructor = 0x1,
+
+    /// <summary>
+    /// Properties and their types should be included in the shape, if any are declared.
+    /// </summary>
+    Properties = 0x2,
```

Comment:
It just occurred to me, but what would be the behavior of either `Constructor` or `Properties` when used with collection or dictionary types? My expectation is that the two should be identical, but does this suggest that we need to pick more generic sounding names for the values?

Similarly, how does this behave when applied to union types? (particularly when one of the union cases is a collection?)

Reply:
The feature only supports object types. I couldn't readily define how this might be used in any other case.

Reply:
I don't have (and at the moment, can't imagine) a use case for any other type than object either. So... if you can think of any change I should make to communicate this through the API, I'd prefer that to expanding scope at this point. I can mention it in the API and concept docs.

Reply:
How does it behave like at the moment? I would presume `None` similarly produces empty shapes, `Full` includes the key/value/element shapes and `Constructor` and `Properties` are no-ops unless targeting union types?

Reply:
In my update yesterday I added API docs to describe current behavior, which is that the enum is ignored for anything except object shapes. 

There's a method in the source gen parser with a massive switch statement (effectively) for each type of shape, and nearly all of them just force `Full` rather than propagate the specified flags. This is because I haven't implemented anything else for each of these other shapes. Collections, enums, tuples,... don't have properties, and not having constructors would break them entirely, so it didn't make sense to me to implement partial shapes.